server/src/test/java/org/elasticsearch/common/geo/SimpleVectorTileFormatterTests.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.common.geo;
+
+import org.elasticsearch.test.ESTestCase;
+import org.hamcrest.Matchers;
+import org.junit.Rule;
+import org.junit.rules.ExpectedException;
+
+import static org.hamcrest.Matchers.equalTo;
+
+public class SimpleVectorTileFormatterTests extends ESTestCase {
+    @Rule
+    public final ExpectedException exception = ExpectedException.none();
+
```

Comment:
I don't think we use this structure to check for exceptions anywhere else in the code base. I would prefer to be consistent, what do you think  if we  move the exception handling to `expectThrows`?

Reply:
Certainly. I did a quick look around and did not see other examples in similar code, so I searched online for favoured options for older versions of junit 4, and this looked like the most common. I'm happy to use whatever is canonical for elastic. However, I am curious why we use such an old version of junit, and do not use the `@Test` and other approaches?

I see `expectThrows` is defined in the lucene code-base, so perhaps upgrading to newer testing like Junit5 and jupiter is very hard at this point?

Reply:
> However, I am curious why we use such an old version of junit, and do not use the @Test and other approaches?

`ESTestCase` extends `LuceneTestCase` so we are bounded to the unit version used in Lucene. I had a look and there seems not to be any proposal to upgrade the library any time soon. I have a vague memory that we do not use @Test annotation for readability and rely on the test methods starting by testXXX.

My vote goes to use the standard `expectThrows` used in the codebase.

Reply:
> I am curious why we use such an old version of junit, and do not use the `@Test` and other approaches?

We're super duper ultra linked into `LuceneTestCase`. And that's linked to the randomized testing library. And those are what locks us into older junit, yeah. It's unfortunate.

For what it's worth, I like `expectThrows` better than most other things I've bumped into that try to do the same thing. It's nice and explicit.

The lack of `@Test` annotation is from the Lucene heritage as well. They've been using the junit-3 style `testFoo` methods for so long and don't really see any value in changing. We're along for the ride. In fact, we forbid the `@Test` annotation, but mostly because it's not worth having two ways to do things.

I also like our randomized testing much more than any other randomized testing I've ever used. quickcheck style things are lovely but come with so much extra complexity in an attempt to try and derive the "smallest" failure case. In my opinion that ain't worth it. Just hand me a failure case and let me pull on it myself to find the smallest one.